packages/core/test/financial-templates/Liquidatable.js

```diff
@@ -173,6 +175,29 @@ contract("Liquidatable", function(accounts) {
     financialContractsAdmin = await FinancialContractsAdmin.deployed();
   });
 
+  const expectNoExcessCollateralToTrim = async () => {
+    let collateralTrimAmount = await liquidationContract.trimExcess.call(collateralToken.address);
+    await liquidationContract.trimExcess(collateralToken.address);
+    let beneficiaryCollateralBalance = await collateralToken.balanceOf(beneficiary);
+
+    assert.equal(collateralTrimAmount.toString(), "0");
+    assert.equal(beneficiaryCollateralBalance.toString(), "0");
+  };
+
+  const expectAndDrainExcessCollateral = async () => {
+    // Drains the collateral from the contract and transfers it all back to the sponsor account to leave the beneficiary empty.
+    await liquidationContract.trimExcess(collateralToken.address);
+    let beneficiaryCollateralBalance = await collateralToken.balanceOf(beneficiary);
+    collateralToken.transfer(sponsor, beneficiaryCollateralBalance.toString(), { from: beneficiary });
+
+    // Assert that nonzero collateral was drained.
+    assert.notEqual(beneficiaryCollateralBalance.toString(), "0");
+  };
+
+  afterEach(async () => {
```

Comment:
This is awesome, does this make the rest of the `expectNoExcessCollateralToTrim()` calls redundant now?

Reply:
Good point. Most of them are probably still necessary because we want to check that there are no intermediate steps that leave untracked tokens sitting in the contract. For instance, I was a little worried that somewhere in the liquidation and dispute flow that a bond or final fee was in the contract but not added to pfc, which would allow someone to drain it using this method. I'll go back and remove any that occur as the last step in a test case, though.